client/frontend/client.go

```diff
@@ -21,155 +21,17 @@
-	ctx, cancel := c.createContext()
-	defer cancel()
-	return c.client.PollForActivityTask(ctx, pollRequest)
-}
-
-func (c *clientImpl) PollForDecisionTask(pollRequest *workflow.PollForDecisionTaskRequest) (*workflow.PollForDecisionTaskResponse, error) {
-	ctx, cancel := c.createContext()
-	defer cancel()
-	return c.client.PollForDecisionTask(ctx, pollRequest)
-}
-
-func (c *clientImpl) RecordActivityTaskHeartbeat(heartbeatRequest *workflow.RecordActivityTaskHeartbeatRequest) (*workflow.RecordActivityTaskHeartbeatResponse, error) {
-	ctx, cancel := c.createContext()
-	defer cancel()
-	return c.client.RecordActivityTaskHeartbeat(ctx, heartbeatRequest)
-}
-
-func (c *clientImpl) RespondDecisionTaskCompleted(request *workflow.RespondDecisionTaskCompletedRequest) error {
-	ctx, cancel := c.createContext()
-	defer cancel()
-	return c.client.RespondDecisionTaskCompleted(ctx, request)
-}
-
-func (c *clientImpl) RespondActivityTaskCompleted(request *workflow.RespondActivityTaskCompletedRequest) error {
-	ctx, cancel := c.createContext()
-	defer cancel()
-	return c.client.RespondActivityTaskCompleted(ctx, request)
-}
-
-func (c *clientImpl) RespondActivityTaskFailed(request *workflow.RespondActivityTaskFailedRequest) error {
-	ctx, cancel := c.createContext()
-	defer cancel()
-	return c.client.RespondActivityTaskFailed(ctx, request)
-}
-
-func (c *clientImpl) RespondActivityTaskCanceled(request *workflow.RespondActivityTaskCanceledRequest) error {
-	ctx, cancel := c.createContext()
-	defer cancel()
-	return c.client.RespondActivityTaskCanceled(ctx, request)
-}
-
-func (c *clientImpl) RequestCancelWorkflowExecution(cancelRequest *workflow.RequestCancelWorkflowExecutionRequest) error {
-	ctx, cancel := c.createContext()
-	defer cancel()
-	return c.client.RequestCancelWorkflowExecution(ctx, cancelRequest)
-}
-
-func (c *clientImpl) SignalWorkflowExecution(request *workflow.SignalWorkflowExecutionRequest) error {
-	ctx, cancel := c.createContext()
-	defer cancel()
-	return c.client.SignalWorkflowExecution(ctx, request)
-}
-
-func (c *clientImpl) TerminateWorkflowExecution(request *workflow.TerminateWorkflowExecutionRequest) error {
-	ctx, cancel := c.createContext()
-	defer cancel()
-	return c.client.TerminateWorkflowExecution(ctx, request)
-}
-
-func (c *clientImpl) ListOpenWorkflowExecutions(
-	listRequest *workflow.ListOpenWorkflowExecutionsRequest) (*workflow.ListOpenWorkflowExecutionsResponse, error) {
-	ctx, cancel := c.createContext()
-	defer cancel()
-	return c.client.ListOpenWorkflowExecutions(ctx, listRequest)
+// Client is the interface exposed by frontend service client
+type Client interface {
+	workflowserviceclient.Interface
 }
 
-func (c *clientImpl) ListClosedWorkflowExecutions(
-	listRequest *workflow.ListClosedWorkflowExecutionsRequest) (*workflow.ListClosedWorkflowExecutionsResponse, error) {
-	ctx, cancel := c.createContext()
```

Comment:
looks like the context is now hidden within the dispatcher. What are the default values for timeouts and retry ? Are they in parity with the old configuration ?

Reply:
No defaults in the dispatcher, the dispatcher will throw error if timeouts are missing, since this is front-end, the caller need to create context with appropriate timeouts like our integration and unit tests. We have default timeouts for history and matching client and they are same as earlier.